website/content/api-docs/secret/kv/kv-v2.mdx

```diff
@@ -146,11 +148,9 @@ have an ACL policy granting the `update` capability.
 
 - `options` `(Map: <optional>)` – An object that holds option settings.
 
-  - `cas` `(int: <optional>)` - Set the "cas" value to use a Check-And-Set
-    operation. If not set the write will be allowed. If set to 0 a write will
-    only be allowed if the key doesn’t exist. If the index is non-zero the
-    write will only be allowed if the key’s current version matches the
-    version specified in the cas parameter.
+- `cas` `(int: <optional>)` - A flag required if Check-And-Set is set to true on
+  either the secret or the engine's config. A write will only then be allowed if 
+  key's current version matches the specified in the cas flag.
```

Comment:
I think everyone might write this differently so I'll just leave my suggestion here 😂 I think yours makes the point but it could be a little more verbose: 

>  `cas` `(int: <optional>)` - This flag is required if `cas_required` is set to `true` on either the secret or the engine's config. In order for a write to be successful, `cas` must be set to the current version of the secret. If `cas` is set to 0, the write will only be allowed if the key doesn't exist. 

I specified `cas_required` and adjusted the wording slightly. I also did keep the 0 case, since that's in the code sample.

Reply:
Well said. I struggled with this one.